packages/components/src/box-control/styles/box-control-styles.ts

```diff
@@ -11,11 +11,18 @@ import BaseUnitControl from '../../unit-control';
 import { rtl } from '../../utils';
 import type { BoxUnitControlProps } from '../types';
 
+const RootWidth = ( {
+	__next40pxDefaultSize,
+}: Pick< BoxUnitControlProps, '__next40pxDefaultSize' > ) => {
+	const maxWidth = __next40pxDefaultSize ? '320px' : '235px';
```

Comment:
Possibly a silly question... I understand why a min-width is required on the inputs, but why do we need a max-width? Can't they just scale?

<img width="1077" alt="Screenshot 2023-11-17 at 17 17 12" src="https://github.com/WordPress/gutenberg/assets/846565/17676e36-4b01-47c7-9c70-bae04801175e">

I guess that's separate...

Inspector aside, shouldn't the component be able to scale down to smaller sizes regardless? 280px (Inspector width) is small, but not _that_ small...

All that said, I don't know if this one component is worth prioritising – I'm not sure it's even used in core.

Apologies if I'm still misunderstanding the question.

Reply:
> why do we need a max-width? Can't they just scale?

I agree that components should rarely have a max-width (it should be the consumer of this component to add one, if needed). @brookewp , could you look into that, and in particular if (by looking at the history of the component) there's a clear reason for having the `max-width` ?



> Inspector aside, shouldn't the component be able to scale down to smaller sizes regardless? 280px (Inspector width) is small, but not that small...
> Apologies if I'm still misunderstanding the question.

Regardless of whether we should be setting a max-width in the component or not, if I understand correctly @brookewp is flagging that, when rendering the `UnitControl` components with the new `40px` height, those inputs need to be wider in order to show their values.

And potentially, when using this component in the editor sidebar, the available width (280px - 2*16px padding) = 248px) may not be enough to have four UnitControls in their "40px" configuration, without cutting the text inside those inputs.

> All that said, I don't know if this one component is worth prioritising – I'm not sure it's even used in core.

I think that BoxControl is still used in the dimensions panel ([see message below](https://github.com/WordPress/gutenberg/pull/56185#issuecomment-1816603164)).

It would be great if we could get a list sorted by priority — although ultimately I seem to remember that all components need to be refactored to the new 40px size. We'd like to wrap up the migration sooner rather than later, so that we can refactor our UIs and ultimately remove both the `__next40pxDefaultSize` prop and the `__unstable-large` size.

Reply:
Gotcha. I think the point stands though – since BoxControl is not _restricted_ to the Inspector, it should scale elegantly to smaller widths? If we wanted to reuse existing patterns, here some options:

### Custom padding (when spacing presets are provided)
<img width="281" alt="Screenshot 2023-11-20 at 10 29 10" src="https://github.com/WordPress/gutenberg/assets/846565/3cb7d916-277e-44e9-a90d-eec785f88fc5">

(The settings button would be omitted).

### Border control
<img width="281" alt="Screenshot 2023-11-20 at 10 31 20" src="https://github.com/WordPress/gutenberg/assets/846565/8ca62476-7877-4f85-ad47-e4a47fb2edd4">

(The color picker would be omitted).

I don't have a strong feeling either way. As I said this component isn't used in core very often since most themes provide spacing presets nowadays. If neither of these patterns will work, or are too difficult to implement, then we'll probably need to revisit the design.

Reply:
>(this can also be tested in storybook itself)

Such a helpful TIL 🥲 Thank you!

>could you look into that, and in particular if (by looking at the history of the component) there's a clear reason for having the max-width ?

This was part of [the initial creation of the component three years ago](https://github.com/WordPress/gutenberg/pull/21492). Nothing was mentioned in relation to the width/input size. The component was made for the Cover block, so my guess is it was sized to fit within the sidebar. So I don't think a hard width is needed. 

>And potentially, when using this component in the editor sidebar, the available width (280px - 2*16px padding) = 248px) may not be enough to have four UnitControls in their "40px" configuration, without cutting the text inside those inputs.

Looks like that's the case; it's not wide enough 😓

Reply:
>As I said this component isn't used in core very often since most themes provide spacing presets nowadays.

So it is likely there for backwards compability? Since it shows in the case when `showSpacingPresetsControl` is `false`. And it was added quite recently in https://github.com/WordPress/gutenberg/pull/48070. 

>since BoxControl is not restricted to the Inspector, it should scale elegantly to smaller widths? If we wanted to reuse existing patterns, here some options:

@ciampo should we look into an alternative pattern when `__next40pxDefaultSize` is enabled?

Reply:
> So it is likely there for backwards compability?

Yes. So rather than coming up with a new design it would be simpler to re-use existing patterns (like the padding / border examples above). That way the component will work nicely in any environment, not just the Inspector. What do you think?

Reply:
I believe the `Custom padding` option makes the most sense for `BoxControl`. However, are we okay with imposing a limitation on the values by including`RangeControl`? cc @ciampo 

If not, are we okay with the `Border control` option when there are only two values? cc @jameskoster 

For example: 

<img width="355" alt="Screenshot 2023-11-23 at 6 38 55 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/3183415e-145f-45d0-a777-105105a4305a">

Reply:
I agree with Brooke, the layout of the custom padding controls feels like a better fit — although that means that we'll have to introduce a slider next to each input.

Reply:
That works for me.